common/src/main/java/com/sequenceiq/cloudbreak/quartz/configuration/SchedulerFactoryConfig.java

```diff
@@ -26,13 +44,28 @@
     @Inject
     private ResourceCheckerJobListener resourceCheckerJobListener;
 
+    @Inject
+    private MeterRegistry meterRegistry;
+
     @Bean
     public SchedulerFactoryBeanCustomizer schedulerFactoryBeanCustomizer() {
         return bean -> {
             bean.setAutoStartup(properties.isAutoSyncEnabled());
             bean.setGlobalJobListeners(resourceCheckerJobListener, jobMetricsListener);
             bean.setGlobalTriggerListeners(triggerMetricsListener);
+            bean.setTaskExecutor(quartzTaskExecutor());
         };
     }
 
+    @Bean(name = QUARTZ_TASK_EXECUTOR)
+    public ExecutorService quartzTaskExecutor() {
+        ThreadPoolTaskExecutor executor = new ThreadPoolTaskExecutor();
+        executor.setCorePoolSize(quartzCorePoolSize);
+        executor.setQueueCapacity(quartzQueueCapacity == 0 ? Integer.MAX_VALUE : quartzQueueCapacity);
```

Comment:
I think this should be always 0, and we should set the maxpoolsize to a reasonable number (same as corepoolsize).

Reply:
As we discussed, I will modify this accordingly